src/components/Layout/index.tsx

```diff
@@ -18,14 +20,40 @@ const navBarContent = (title: string, refUrl: string, isActive: boolean = false)
     </Link>
   );
 };
-
+const { SUCCESS, ERROR } = ToastTypes;
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
 const Layout: FC<Props> = ({ children }) => {
+  const [IsUserAuthorized, setIsUserAuthorized] = useState(false);
+
+  useEffect(() => {
```

Comment:
Move this to a separate function (util). Keep the component only for the single task it is responsible for